typescript/core/src/parse.ts

```diff
@@ -62,10 +62,15 @@ export function parseRecord({
   }
 
   if (!isKnownOpcode(opcode)) {
+    const data = new Uint8Array(
+      view.buffer,
+      view.byteOffset + headerReader.offset,
+      recordLengthNum,
+    ).slice();
     const record: TypedMcapRecord = {
       type: "Unknown",
       opcode,
-      data: new Uint8Array(view.buffer, view.byteOffset + headerReader.offset, recordLengthNum),
```

Comment:
This did not previously copy the buffer - do we want to be doing that copy now?

Reply:
I updated this to restore the original behavior for comparison reasons, but I believe this should be a copy.